app/main.py

```diff
@@ -1,24 +1,112 @@
-from fastapi import FastAPI, HTTPException, status
+from datetime import datetime
+from fastapi import FastAPI, HTTPException, status, File, UploadFile
+from PIL import Image
+import io
+import torch
 
+from fastapi.staticfiles import StaticFiles
+
+from Exceptions.PlantsUndetectedError import PlantsUndetectedError
+from Util import FileHandler
+from Util.Data import RawEntry, ProcessedEntry, load_config, connect, insert_raw_entry, insert_processed_entry
+from Models.HumanDetection.HumanDetector import Classifier as HumanDetector
+from Util.PlantDetector import detect
+from app.Messages import StatusEnum, MessageResponse, PlantGrowthDataResponse
+
+# START OPTIONS
+CHECK_HUMANS = False  # Verify if humans are in the image
+
+# Set up app
 app = FastAPI()
+app.mount("/ui", StaticFiles(directory="app/static", html=True), name="static")
+
+# Set up PostgreSQL
+config = load_config()
+conn = connect(config)
+
+# Get device for ML
+device = "cpu"
+if torch.backends.mps.is_available():
+    device = "mps"
+if torch.cuda.is_available():
+    device = "cuda"
+
+# Load Models
+human_detector = HumanDetector(pretrained=True)
+human_detector = human_detector.to(device)
+
 
 @app.get("/")
 async def hello_world():
     return {"response": "Hello World"}
 
-@app.get("/upload_images")
-async def upload_images():
-    raise HTTPException(
-        status_code=status.HTTP_501_NOT_IMPLEMENTED
+
+@app.post("/upload_image")
+async def upload_image(file: UploadFile = File(...)):
+
+    # This try catch clause is shamelessly stolen from:
+    # https://stackoverflow.com/questions/73810377/how-to-save-an-uploaded-image-to-fastapi-using-python-imaging-library-pil
+    try:
+        im = Image.open(file.file)
+        if im.mode in ("RGBA", "P"):
+            im = im.convert("RGB")
+        buf = io.BytesIO()
+        im.save(buf, 'JPEG', quality=50)
+        # to get the entire bytes of the buffer use:
+        contents = buf.getvalue()
+        # or, to read from `buf` (which is a file-like object), call this first:
+        buf.seek(0)  # to rewind the cursor to the start of the buffer
+    except Exception:
+        raise HTTPException(status_code=500, detail='Something went wrong')
+    finally:
+        file.file.close()
+        buf.close()
+
+    # Check For Humans
+    img_normalised = human_detector.tensor_transform(im)
+    img_normalised = img_normalised.unsqueeze_(0)
+    img_normalised = img_normalised.to(device)
+    output = human_detector(img_normalised)
+    output = torch.argmax(output.detach().cpu())
+    if output == 1 and CHECK_HUMANS:
```

Comment:
Can save time and compute power by reading `CHECK_HUMANS` before we do any of this

Reply:
That's a great idea, updated